proto/umee/leverage/v1beta1/genesis.proto

```diff
@@ -2,12 +2,50 @@ syntax = "proto3";
 package umeenetwork.umee.leverage.v1beta1;
 
 import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
 import "umee/leverage/v1beta1/leverage.proto";
 
 option go_package = "github.com/umee-network/umee/x/leverage/types";
 
 // GenesisState defines the x/leverage module's genesis state.
 message GenesisState {
-  Params         params   = 1 [(gogoproto.nullable) = false];
-  repeated Token registry = 2 [(gogoproto.nullable) = false];
+  Params                            params              = 1 [(gogoproto.nullable) = false];
+  repeated Token                    registry            = 2 [(gogoproto.nullable) = false];
+  repeated Borrow                   borrows             = 3 [(gogoproto.nullable) = false];
+  repeated CollateralSetting        collateral_settings = 4 [(gogoproto.nullable) = false];
+  repeated Collateral               collateral          = 5 [(gogoproto.nullable) = false];
+  repeated cosmos.base.v1beta1.Coin reserves            = 6
+      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+  int64                                last_interest_time = 7;
+  repeated cosmos.base.v1beta1.DecCoin exchange_rates = 8
+      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.DecCoins"];
+  repeated BadDebt bad_debts = 9 [(gogoproto.nullable) = false];
+  repeated cosmos.base.v1beta1.DecCoin borrow_APYs = 10
+      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.DecCoins"];
+  repeated cosmos.base.v1beta1.DecCoin lend_APYs = 11
+      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.DecCoins"];
```

Comment:
Note: For these three fields using `sdk.DecCoins`, I'd like to point out that the `Amount` does not store an amount of coins, but rather the decimal quantity in question.

For example, inside `borrow_APYs` the following would mean an interest rate of 35%, not 0.35 uatom

> sdk.Dec{
>   Denom: "uatom",
>   Amount: 0.35,
> }

Is this an acceptable use of the built-in types, or should I create three message types `BorrowAPY`, `LendAPY`, and `ExchangeRate` that each contain `denom` and an `sdk.Dec`?

Reply:
Good point, I would **not** use the `sdk.DecCoin` type here, instead define concrete types to contain these values as you've suggested.

Reply:
- [x] Replaced with concrete types